### DISCUSSION TITLE
Request for Report: Income_Expenses_Debits Payment_Networth-ByYear

### DISCUSSION THREAD

**Discussion by learingmmex** [moneymanagerex/moneymanagerex#5323] [Q&A]
# Description of the feature
it would be nice to have report showing yearly the Income, Expenses, Debits Payment (To Loans and Credit Cards payments), and Networth
Option to hide/show Graph is also preferred when the list of years lengthy

Please describe your feature with details.

![Capture](https://user-images.githubusercontent.com/92601421/201318120-994bfecf-a924-4749-986a-a08a16545c0b.PNG)

**Comment by whalley** [comment]
> it would be nice to have report showing yearly the Income, Expenses, Debits Payment (To Loans and Credit Cards payments), and Networth

So, the Income vs Expenses - Monthly report is probably the closest to this, and having a yearly version of it should be straight-forward.  

I'm interested to understand what you mean by 'Debits Payment (To Loans and Credit Cards payments)'. Are these, in your view, transfers to the credit/loan accounts t pay the expenses already built up in say credit card transactions? In your calculations of 'Cumulative' these seem to be classed as a form of 'Expense', so I'm not sure.

**Comment by learingmmex** [comment]
So, the Income vs Expenses - Monthly report is probably the closest to this, and having a yearly version of it should be straight-forward.
*****_Answer: _yes, you are correct Income vs Expenses is the closest one_**

I'm interested to understand what you mean by 'Debits Payment (To Loans and Credit Cards payments)'. Are these, in your view, transfers to the credit/loan accounts t pay the expenses already built up in say credit card transactions? 
****_Answer: yes, as you said_**, i mean by 'Debits Payment (To Loans and Credit Cards payments)' are transfers to the credit/loan accounts to pay the expenses already built up in credit card transactions**_

In your calculations of 'Cumulative' these seem to be classed as a form of 'Expense', so I'm not sure.
_****_Answer: maybe i did not presented correctly, i mean the cumulative what is left from the income after we pay Loans/Credit Cards and other expenses**_

**Comment by whalley** [comment]
> ****Answer: yes, as you said, i mean by 'Debits Payment (To Loans and Credit Cards payments)' are transfers to the credit/loan accounts to pay the expenses already built up in credit card transactions_
>In your calculations of 'Cumulative' these seem to be classed as a form of 'Expense', so I'm not sure.
_****Answer: maybe i did not presented correctly, i mean the cumulative what is left from the income after we pay Loans/Credit Cards and other expenses**

So, given your figures for 2020

- Income = 348,000 
- Expenses = 200,000 (which will include purchases made on credit cards as you have confirmed)
- Debits Paid = 100,000 (aren't these just transfers to the credit card accounts, which from a financial worth perspective are already included in Expenses)
- Then you have cumulative as Income - Expenses - Debits Paid = 48,000. Not sure why this is not just Income - Expenses.

I'm obviously missing something here 😕

**Comment by learingmmex** [comment]
example 1: 
i receive my salary 100$ in my checking account AAA, this is recorded as income
when i pay for lunch from Credit Card 50$, then this is recorded as expense. simultaneously my liability to the credit card increases by 50$.

to get rid of this liability (50$), i have to transfer 50$ from my checking account AAA to the credit card, then the balance of the credit card become zero.

then if i consider paying debit is expense, then the total expense = 50$ which i paid for lunch + 50$ which i paid to the credit card = 100$

**Comment by learingmmex** [comment]
additional note: there is report in the General Report "Income_Expenses-DepositVsExpensesAndLoanRepaymentByMonth"

it shows the monies transferred to the Loan accounts. it would solve my issue if the transfer is also considered to Credit Card accounts.

i tried my self to change the code, but i am not good in coding

**Comment by whalley** [comment]
> then if i consider paying debit is expense

But why consider it like that? Paying debt on your Credit Card is just a transfer of funds within your financial worth. The net result is no different than if you had paid for the lunch using your checking account AAA.

In your original numbers (for 2020) if you income is 348,000 and your expenses is 200,000 then the cumulative net for the year is 148,000 and NOT 48,000 as you have it.

**Comment by tactilis** [comment]
@learingmmex 
>> then if i consider paying debit is expense

@whalley
> But why consider it like that? Paying debt on your Credit Card is just a transfer of funds within your financial worth. The net result is no different than if you had paid for the lunch using your checking account AAA.

Agreed.
 
 

@learingmmex you may like to read these two threads on the MMEX forum. They describe how you should handle credit cards.

[https://forum.moneymanagerex.org/viewtopic.php?p=22054](https://forum.moneymanagerex.org/viewtopic.php?p=22054)

[https://forum.moneymanagerex.org/viewtopic.php?p=22008](https://forum.moneymanagerex.org/viewtopic.php?p=22008)

If, after studying these threads, you have any questions then please do ask again here.

**Comment by learingmmex** [comment]
thank you for your response, however, i think i wasn't clear enough in my explanation. As i said there is report in the General Report (.grm) "Income_Expenses-DepositVsExpensesAndLoanRepaymentByMonth" i modified it to include credit cards payments, but i did not know how include the currency conversion (i am using more than one currency), i appreciate if you can help on that and my issue will be solved.

the code:

SELECT 'month' AS periode_name,
       cadata.periode AS periode,
       sum(cadata.Deposit) AS Deposit,
       sum(cadata.Withdrawal) AS Withdrawal,
       sum(cadata.Transfer) AS Transfer,
       round(sum(cadata.Deposit) + sum(cadata.Withdrawal) + sum(cadata.Transfer), 2) AS Total,
       (
           SELECT sum(al3.INITIALBAL) 
             FROM accountlist_V1 AS al3
       )
       AS initialbal
  FROM (
           SELECT strftime('%Y-%m', ca.TRANSDATE) AS periode,
                  CASE WHEN ca.transcode = 'Deposit' THEN ca.transamount ELSE 0 END AS Deposit,
                  CASE WHEN ca.transcode = 'Withdrawal' THEN -ca.transamount ELSE 0 END AS Withdrawal,
                  CASE WHEN ca.transcode = 'Transfer' AND 
                            ca.toaccountid IN (
                          SELECT al1.accountid
                            FROM accountlist_V1 AS al1
                           WHERE al1.ACCOUNTTYPE IN ('Loan', 'Credit Card')
                      )
                  THEN -ca.transamount WHEN ca.transcode = 'Transfer' AND 
                                            ca.accountid IN (
                          SELECT al2.accountid
                            FROM accountlist_V1 AS al2
                           WHERE al2.ACCOUNTTYPE IN ('Loan', 'Credit Card')
                      )
                  THEN ca.transamount ELSE 0 END AS Transfer
             FROM checkingaccount_V1 AS ca
            WHERE ca.STATUS NOT IN ('D', 'V')-- and TRANSDATE > date('now', 'start of month','-5 year','localtime') 
       )
       AS cadata
 GROUP BY periode
 ORDER BY periode ASC;

**Comment by whalley** [comment]
The GRM report `Account-AccountBalances` includes logic to do currency conversions that you may want to use.
